x-pack/plugins/enterprise_search/server/lib/ml_inference_pipeline/get_inference_pipelines.ts

```diff
@@ -0,0 +1,94 @@
+ * a model inaccessible in the current Kibana space.
+ * @param esClient the Elasticsearch Client to use to fetch the errors.
+ * @param trainedModelsProvider ML trained models provider.
+ */
+export const getMlInferencePipelines = async (
+  esClient: ElasticsearchClient,
+  trainedModelsProvider: MlTrainedModels | undefined
+): Promise<Record<string, IngestPipeline>> => {
+  if (!trainedModelsProvider) {
+    return Promise.reject(new Error('Machine Learning is not enabled'));
+  }
+
+  // Fetch all ML inference pipelines
+  const fetchedInferencePipelines = await esClient.ingest.getPipeline({
+    id: 'ml-inference-*',
+  });
+
+  // Fetch all trained models from the current Kibana space, and identify those that have their
+  // model_id redacted
+  const trainedModels = await getMlModelConfigsForModelIds(esClient, trainedModelsProvider);
+  const redactedModelIds = Object.values(trainedModels)
+    .filter((modelConfig) => !modelConfig.modelId)
+    .map((modelConfig) => modelConfig.trainedModelName);
+
+  // Process pipelines: check if the model_id is one of the redacted ones, if so, redact it in the
+  // result as well
+  const inferencePipelinesResult: Record<string, IngestPipeline> = {};
+  Object.keys(fetchedInferencePipelines).forEach((name) => {
+    const inferencePipeline = fetchedInferencePipelines[name];
+
+    inferencePipelinesResult[name] = isModelIdRedacted(inferencePipeline, redactedModelIds)
+      ? redactModelId(inferencePipeline)
+      : inferencePipeline;
+  });
+
+  return Promise.resolve(inferencePipelinesResult);
+};
+
+/**
+ * Convenience function that finds the inference processor in a given ingest pipeline.
+ * @param pipeline the pipeline.
+ */
+function getInferenceProcessor(pipeline: IngestPipeline): IngestProcessorContainer | undefined {
+  const processors = pipeline.processors || [];
+
+  return processors.find((processor) => processor.hasOwnProperty('inference'));
+}
+
+/**
+ * Convenience function for evaluating whether the trained model the supplied pipeline references
+ * is redacted.
+ * @param pipeline the pipeline.
+ * @param redactedModelIds Array of known redacted model IDs.
+ */
+function isModelIdRedacted(pipeline: IngestPipeline, redactedModelIds: string[]) {
+  const inferenceProcessor = getInferenceProcessor(pipeline);
+
+  return (
+    inferenceProcessor?.inference &&
+    redactedModelIds.includes(inferenceProcessor.inference.model_id)
+  );
+}
+
+/**
+ * Convenience function that redacts the trained model ID in a given ingest pipeline by setting it
+ * to `model_id: ''`.
+ * @param pipeline the pipeline to process.
+ * @returns a copy of the input pipeline with the model ID redacted.
+ */
+function redactModelId(pipeline: IngestPipeline): IngestPipeline {
+  const modifiedPipeline = { ...pipeline };
+
+  const inferenceProcessor = getInferenceProcessor(modifiedPipeline);
+  if (inferenceProcessor?.inference) {
+    inferenceProcessor.inference.model_id = '';
```

Comment:
`model_id` is `string` so we cannot make it `undefined` here. I chose to set it to an empty string instead.

Reply:
can a string be set to `null` instead of the empty string with this definition? That might be a bit clearer

Reply:
No 😞
_Type 'null' is not assignable to type 'string'_

Reply:
That's disappointing :( - empty string it is then :)